crates/codegen/parser/runtime/src/napi_interface/cst.rs

```diff
@@ -68,10 +68,45 @@ impl RuleNode {
             .into()
     }
 
+    #[napi(catch_unwind, js_name = "toJSON")]
+    /// Serialize the token node to JSON.
+    ///
+    /// This method is intended for debugging purposes and may not be stable.
```

Comment:
Given that this roughly maps to our actual properties/APIs, with the same keys/values, I think it is as stable as other properties. I wonder why do we need the extra warning here?

Reply:
>  We can discuss further what we want to explore as part of having pure data external interface (JSON) for Slang, whether that's exporting or importing JSON, but I'd consider this a separate issue as it requires more thought and design; the original issue, from what I understand, was primarily about improving the debugging/inspection experience.

I think of this similar to how we expose `serde(Deserialize)` on the Rust types, and the keys/values follow roughly the same shape. It is part of our public API (and also CLI's `--json` output). Not sure if this would be different.

Reply:
Although for thoroughness, it is worth noting that they would have different casing for field names between languages, and we should make a decision on whether we want to consolidate them.

Reply:
We discussed this at some point (Q1 planning?) and IIRC we established that this issue is more about introspection for debugging purposes and that we do not want to bless this as a stable format that people should expect not to change, at least for the time being, so that's exactly why I added this wording here.

Reply:
Discussed f2f: since this matches the same JSON we provide from the Rust API and the CLI, then it is as "stable" as the other ones get.
I suggest removing the warning if so.

Reply:
Already done in https://github.com/NomicFoundation/slang/pull/908/commits/f0552aacb0879f5e56c1f75d1490b9fe567ca52d